src/installer/tests/HostActivation.Tests/FrameworkResolution/ComplexHierarchies.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class ComplexHierarchies :
+        FrameworkResolutionBase,
+        IClassFixture<ComplexHierarchies.SharedTestState>
+    {
+        private SharedTestState SharedState { get; }
+
+        public ComplexHierarchies(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        public class SharedTestState : SharedTestStateBase
+        {
+            public TestApp FrameworkReferenceApp { get; }
+
+            public DotNetCli DotNetWithMultipleFrameworks { get; }
+
+            public SharedTestState()
+            {
+                DotNetWithMultipleFrameworks = DotNet("DotNetWithMultipleFrameworks")
+                    .AddMicrosoftNETCoreAppFrameworkMockHostPolicy("5.1.1")
+                    .AddFramework("MiddleWare", "2.1.2", runtimeConfig =>
+                        runtimeConfig.WithFramework(MicrosoftNETCoreApp, "5.1.1"))
+                    .AddFramework("SerializerWare", "3.0.1", runtimeConfig =>
+                        runtimeConfig
+                            .WithFramework(MicrosoftNETCoreApp, "5.1.0")
+                            .WithFramework("MiddleWare", "2.1.0"))
+                    .AddFramework("OMWare", "7.3.1", runtimeConfig =>
+                        runtimeConfig
+                            .WithFramework(MicrosoftNETCoreApp, "5.1.0")
+                            .WithFramework("MiddleWare", "2.1.0"))
+                    .Build();
+
+                FrameworkReferenceApp = CreateFrameworkReferenceApp();
+            }
+        }
+
+        [Fact]
+        public void TwoAppFrameworksOnTopOfMiddleWare()
```

Comment:
It would also be an issue with just one framework on top of MiddleWare, as long as it lists MiddleWare last instead of NETCore.App, right? Or does is need the two both doing that?

Reply:
Yes - it probably would fail with just one... I added two as it's closer to the original customer reported repro.